replacy/__init__.py

```diff
@@ -157,12 +183,25 @@ def cb(matcher, doc, i, matches):
 
             span._.description = self.match_dict[match_name].get("description", "")
             span._.category = self.match_dict[match_name].get("category", "")
+            span._.subcategory = self.match_dict[match_name].get("subcategory", "")
+            span._.suggestions_separator = self.match_dict[match_name].get("suggestions_separator", " ")
+            span._.construct_suggestion_function = self.match_dict[match_name].get("construct_suggestion_function", "")
             for novel_prop, default_value in self.novel_prop_defaults.items():
                 setattr(
                     span._,
                     novel_prop,
                     self.match_dict[match_name].get(novel_prop, default_value),
                 )
+
+            if self.extended_span:
+                span.match_name = span._.match_name
+                span.description = span._.description
+                span.category = span._.category
+                span.subcategory = span._.subcategory
+                span.suggestions = span._.suggestions
+                span.suggestions_separator = span._.suggestions_separator
+                span.construct_suggestion_function = span._.construct_suggestion_function
+
```

Comment:
okay so in `ESpan` we add a factory method for creating `ESpans`, and give them exactly the (writer-specific) properties we need. but that makes this only usable for us, and means replaCy needs a code change if extension properties change. But replaCy tries to be useful for not-Writer projects, and also tries to be dynamic so that if you add a key to the match_dict, it ends up a property of the matched span.

I think if we use the definition I propse this isn't an issue and we don't need to manually set these